xbmc/platform/win32/WinMain.cpp

```diff
@@ -84,26 +99,43 @@ INT WINAPI WinMain(HINSTANCE hInst, HINSTANCE, LPSTR commandLine, INT)
   using KODI::PLATFORM::WINDOWS::ToW;
   std::string appName = CCompileInfo::GetAppName();
   HANDLE appRunningMutex = CreateMutex(nullptr, FALSE, ToW(appName + " Media Center").c_str());
-  if (GetLastError() == ERROR_ALREADY_EXISTS)
+  if (appRunningMutex == nullptr)
```

Comment:
This code change is entirely wrong to check `GetLastError() == ERROR_ALREADY_EXISTS` appRunningMutex is valid HANDLE then is not possible `appRunningMutex == nullptr` and `GetLastError() == ERROR_ALREADY_EXISTS` at same time.

CreateMutex succeeds (!nullptr) and GetLastError() == ERROR_ALREADY_EXISTS when other instance of Kodi is already running.

Reply:
Good call, I switched a couple things. Also noticed that ReleaseMutex at the end returns in error, because mutex ownership is not claimed during creation and it's not waited on either. I'll remove.
What's needed instead is a CloseHandle().